sp_api/api/shipping/shipping.py

```diff
@@ -0,0 +1,261 @@
+        "addressLine2": "enim",
+        "addressLine3": "sed ullamco qui tempor",
+        "email": "est",
+        "copyEmails": [
+            "amet",
+            "cupidatat"
+        ],
+        "phoneNumber": "Lo"
+    },
+    "shipDate": "1985-10-27T16:16:28.801Z"
+}
+    """
+
+    @sp_endpoint('/shipping/v1/purchaseShipment', method='POST')
+    def purchase_shipment(self, data, **kwargs) -> ApiResponse:
+        # TODO why do they blend the kwargs in?
+        return self._request(kwargs.pop('path'), data=data)
+
+    """
+    {
+    "containerSpecifications": [
+        {
+            "dimensions": {
+                "height": 55820746.24605319,
+                "length": -17011895.59158598,
+                "unit": "CM",
+                "width": 92714119.67221498
+            },
+            "weight": {
+                "unit": {
+                    "value": "<Error: Too many levels of nesting to fake this schema>"
+                },
+                "value": {
+                    "value": "<Error: Too many levels of nesting to fake this schema>"
+                }
+            }
+        },
+        {
+            "dimensions": {
+                "height": 52794069.63785118,
+                "length": 38093725.6342161,
+                "unit": "IN",
+                "width": -69636208.8350734
+            },
+            "weight": {
+                "unit": {
+                    "value": "<Error: Too many levels of nesting to fake this schema>"
+                },
+                "value": {
+                    "value": "<Error: Too many levels of nesting to fake this schema>"
+                }
+            }
+        }
+    ],
+    "serviceTypes": [
+        "Amazon Shipping Premium",
+        "Amazon Shipping Ground"
+    ],
+    "shipFrom": {
+        "addressLine1": "ad eu",
+        "city": "id in cupidatat officia",
+        "countryCode": "ex",
+        "name": "consectetur",
+        "postalCode": "com",
+        "stateOrRegion": "laborum veniam",
+        "addressLine2": "ex eu consectetur minim",
+        "addressLine3": "deserunt incididunt nisi",
+        "email": "culpa e",
+        "copyEmails": [
+            "deserunt Duis Except",
+            "ad ex"
+        ],
+        "phoneNumber": "veniam in ir"
+    },
+    "shipTo": {
+        "addressLine1": "cupidatat elit laborum",
+        "city": "consectetur anim magna sint",
+        "countryCode": "no",
+        "name": "ad ",
+        "postalCode": "culpa labore",
+        "stateOrRegion": "labore laborum qui est",
+        "addressLine2": "enim qui tempor deserunt ut",
+        "addressLine3": "fugiat pariatur el",
+        "email": "non",
+        "copyEmails": [
+            "aute adipisicing sunt",
+            "nostru"
+        ],
+        "phoneNumber": "Duis ad ei"
+    },
+    "shipDate": "2001-05-19T22:20:47.778Z"
+}
+    """
+    @sp_endpoint('/shipping/v1/rates', method='POST')
+    def purchase_shipment(self, data, **kwargs) -> ApiResponse:
```

Comment:
issue: this method is repeated. Maybe it could be rates.

Reply:
indeed, I'm still working on it and validating it on the thing the API is being used on right now